tensorflow-core/tensorflow-core-native/src/main/native/org/tensorflow/internal/c_api/graph.h

```diff
@@ -0,0 +1,40 @@
+/* Copyright 2024 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_JAVA_GRAPH_H_
+#define TENSORFLOW_JAVA_GRAPH_H_
+
+#include "tensorflow/c/c_api.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/// Unique identifier of a TensorFlow graph instance
+typedef void* TF_GraphId;
+
+/// Returns the unique identifier of the graph `g`
+TF_CAPI_EXPORT extern TF_GraphId TF_GetGraphId(const TF_Graph* g);
+
+/// Remove an operation from the name map of the graph `g`, so that it cannot be reversely looked up by name.
+/// This is particularly useful for preventing custom gradient operations to pollute the graph namespace.
+TF_CAPI_EXPORT extern void TF_UnmapOperationName(TF_Graph* g, TF_Operation* operation);
+
+#ifdef __cplusplus
+} /* end extern "C" */
+#include "graph_impl.cc"
```

Comment:
My C++ is very rusty, so is it worth a comment as to why we're including the cc file into the header?

Reply:
That's the only way I found to include custom C++ code during JavaCPP compilation. @saudet , is there another way to provide non-headers files to be compiled with the JNI code other than including it as part of a header that is parsed?

Though I was thinking also that this could be useful if we want to exclude these CC files from the Windows build (as we still don't know if the dependent C++ API headers will be available).

Reply:
We could also do that by putting those files in `@Platform(include=...` but removing them in `init()` only when called from the Parser, although I don't think we'd have any problems just leaving them there, but anyway if you did those kinds of hacks are mostly for cases when we don't want to modify the original code. The way you did it is fine

Reply:
So @Craigacp you're also fine with this, I suggest we keep it like that for now

Reply:
Sure, but it would be good to comment that it's due to the way JavaCPP builds it, in case we eventually get more control over that build so we can go back to a more traditional C++ layout.